File: chess-game/chess-game.v3/chessapp/interface/client/chess-app/src/app/pages/game/play/play/models/pieces/knight.ts
import { Cell } from "../board/ cell";
import { Side } from "../side";
import { Piece } from "./piece";
import { PieceType } from "./piece_type";

export class Knight extends Piece {
    override get can_move_over(): boolean {
        return true;
    }
    
    constructor(piece_id: string, side: Side) {
        super(piece_id, side, PieceType.Knight);
    }

    override validateMove(from: Cell, to: Cell): boolean {
        const [delta_file, delta_rank] = this.calculateMoveDeltas(from, to);
        let tuple = [Math.abs(delta_file), Math.abs(delta_rank)];

        const allowedPoints: [number, number][] = [
            [1, 2],
            [2, 1],
          ];

        return allowedPoints.some(allowed => allowed[0] === tuple[0] && allowed[1] === tuple[1]);
    }
}

File: backend/chessapp/domain/pieces/knight.py
from ...domain.pieces.piece import Piece
from ...domain.pieces.piece_type import PieceType
from ...domain.value_objects.side import Side

class Knight(Piece):

    def get_points(self):
        return 30

    def __init__(self, side: Side):
        super().__init__(side, PieceType.Knight)